### PR TITLE
Added Connection Broker support and test script

### PR DESCRIPTION
**Pull request by steevaavoo**

### CODE REVIEW

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 1]
```diff
@@ -48,6 +48,9 @@ function Get-sbRDSession {
     param(
         [CmdletBinding()]
         [Parameter(Mandatory = $false)]
+        [Alias("RdsServer")]
+        [string]$ConnectionBroker,
+        [Parameter(Mandatory = $false)]
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:53]: Make sure you leave a blank line between params; easier to read.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 2]
```diff
@@ -67,8 +74,31 @@ function Get-sbRDSession {
         [string[]]$UserName = $null
     )
 
+    Begin {
+        if (-not($ConnectionBroker)) {
+            Write-Verbose "[BEGIN  :] No Connection Broker specified, querying and connecting to local host FQDN."
+            $computerinfo = Get-CimInstance -ClassName Win32_ComputerSystem
+            $ConnectionBroker = "$($computerInfo.DNSHostName).$($computerInfo.Domain)"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:81]: Like it!

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 3]
```diff
@@ -67,8 +74,31 @@ function Get-sbRDSession {
...
+        } else {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:82]: Is this blank `else {}` meant to be here?

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 4]
```diff
@@ -67,8 +74,31 @@ function Get-sbRDSession {
...
+        }
+
+        if (-not($SkipCheck)) {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Avoid using `-not` if you can, as it's easier for people to follow flow.
Consider using:
```
if ($SkipCheck.IsPresent) { }
```

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 5]
```diff
@@ -125,7 +159,7 @@ Update-TypeData -TypeName $myType -MemberType ScriptMethod -MemberName SendMessa
         MessageTitle     = $MessageTitle
         MessageBody      = $MessageBody
     }
-    Write-Verbose "Sending message to $($session.UserName)"
+    Write-Verbose "[PROCESS:] Sending message to $($session.UserName)"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:162]: Nice logging throughout :)

**Review by adamrushuk — CHANGES_REQUESTED** [review]
Looks decent, just a few minor things

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 1 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:53]: Ah yes. Thanks and done!

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 3 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:82]: It is not :|

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 4 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Understood - will adjust. On the `connectionbroker` example above - would you agree that's a more appropriate use of it?

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 5 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:162]: `[REVIEW :] Learned from the best!`

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 2 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:81]: Thank you sir!

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 4 above)
Comment (adamrushuk) [reply ManageRDSessions/ManageRDSessions.psm1]: yeah that's fine

**Comment by adamrushuk** [comment]
LGTM

**Review by adamrushuk — APPROVED** [review]

**Pull request merged**